### PR TITLE
MAUI-775: Custom report for Tonga Covid Raw Data

### PR DESCRIPTION
**Pull request by chris-pollard**
### Changes:
- Add one report solely in javascript
- Add another report with same output but using a raw sql execution.

### CODE REVIEW

**Comment by chris-pollard** [comment]
Note that there are two versions here; one is javascript only, the other uses .executeSql() to directly query the database using the existing sql code for the dagster report. I did this to compare performance, but turns out there's not a huge difference. javascript is approx 31-32 seconds, sql is 29 seconds. I'd like some advice here, as to how to manage this within the timeout period.

**Review by rohan-bes — CHANGES_REQUESTED** [review]
Looks pretty good @chris-pollard! Can't wait to have this one in! Happy to pair on the performance issues together, but my first recommendation would be to look for areas where async logic can be parallelised 👍

File: packages/report-server/src/reportBuilder/customReports/tongaCovidRawData.ts [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+  programCode: string;
+  dataElementCodes: string[];
+}
+
+interface AncestorData {
+  villageByIndividual: Record<string, Record<string, string>>;
+  islandByIndividual: Record<string, string>;
+}
+
+const NOW = new Date();
```
[the remark this answers is not in the document]
Comment (chris-pollard) [reply packages/report-server/src/reportBuilder/customReports/tongaCovidRawData.ts:31]: Ah, no, I definitely didn't mean that.

File: packages/report-server/src/reportBuilder/customReports/tongaCovidRawData.ts [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+      undefined,
+      individualsByCode,
+      hierarchy,
+      { startDate, endDate, period },
+      dataElementCodes,
+    );
+  };
+
+  const registrationEvents = await fetch(registrationOptions);
+  const resultsEvents = await fetch(resultsOptions);
```
[the remark this answers is not in the document]
Comment (chris-pollard) [reply packages/report-server/src/reportBuilder/customReports/tongaCovidRawData.ts:135]: Yes, good idea. I implemented this, and it saves about 4 seconds.

**Comment by rohan-bes** [comment]
Closing in favour of https://github.com/beyondessential/tupaia/pull/4176

**Pull request closed without merging**